src/decider/agent/BulkReuser.php

```diff
@@ -51,34 +51,52 @@ public function rerunBulkAndDeciderOnUpload($uploadId, $groupId, $userId, $bulkI
      * UploadDao from container
      */
     $uploadDao = $GLOBALS['container']->get('dao.upload');
-    $topItem = $uploadDao->getUploadParent($uploadId);
-    /** @var DeciderJobAgentPlugin $deciderPlugin
-     * DeciderJobAgentPlugin object to add deciderjob
-     */
-    $deciderPlugin = plugin_find("agent_deciderjob");
-    $dependecies = array();
-    $sql = "INSERT INTO license_ref_bulk (user_fk,group_fk,rf_text,upload_fk,uploadtree_fk,ignore_irrelevant,bulk_delimiters) "
-            . "SELECT $1 AS user_fk, $2 AS group_fk,rf_text,$3 AS upload_fk, $4 as uploadtree_fk, ignore_irrelevant, bulk_delimiters
-              FROM license_ref_bulk WHERE lrb_pk=$5 RETURNING lrb_pk, $5 as lrb_origin";
-    $sqlLic = "INSERT INTO license_set_bulk (lrb_fk, rf_fk, removing, comment, reportinfo, acknowledgement) "
-            ."SELECT $1 as lrb_fk, rf_fk, removing, comment, reportinfo, acknowledgement FROM license_set_bulk WHERE lrb_fk=$2";
-    $this->dbManager->prepare($stmt=__METHOD__.'cloneBulk', $sql);
-    $this->dbManager->prepare($stmtLic=__METHOD__.'cloneBulkLic', $sqlLic);
-    $res = $this->dbManager->execute($stmt,array($userId,$groupId,$uploadId,$topItem, $bulkId));
-    $row = $this->dbManager->fetchArray($res);
-    $this->dbManager->freeResult($res);
-    $resLic = $this->dbManager->execute($stmtLic,array($row['lrb_pk'],$row['lrb_origin']));
-    $this->dbManager->freeResult($resLic);
-    $upload = $uploadDao->getUpload($uploadId);
-    $uploadName = $upload->getFilename();
-    $job_pk = \JobAddJob($userId, $groupId, $uploadName, $uploadId);
-    $dependecies = array(array('name' => 'agent_monk_bulk', 'args' => $row['lrb_pk']));
-    $errorMsg = '';
-    $jqId = $deciderPlugin->AgentAdd($job_pk, $uploadId, $errorMsg, $dependecies);
+    $nTopItem = $uploadDao->getUploadParent($uploadId);
+    $pUTree = $this->dbManager->getSingleRow("SELECT uploadtree_fk FROM license_ref_bulk WHERE lrb_pk=$1",
+      array($bulkId), __METHOD__);
+    $uploadEntry = $uploadDao->getUploadEntry($pUTree['uploadtree_fk']);
+    $pUID = intval($uploadEntry['upload_fk']);
+    $pTopItem = $uploadDao->getUploadParent($pUID);
+    if ($pTopItem == $pUTree['uploadtree_fk']) {
+      $topItem = $nTopItem;
+    } else {
+      $pfUTree = $this->dbManager->getSingleRow("SELECT uploadtree_pk FROM uploadtree WHERE upload_fk=$1 AND ufile_name=$2 AND ufile_mode=$3",
+        array($pUID, $uploadEntry['ufile_name'], $uploadEntry['ufile_mode']), __METHOD__.'getRealUploadtreeEntry');
+      if (!empty($pfUTree) && count($pfUTree) <= 1) {
```

Comment:
If the above comment is to be considered, the conditions here should flip as well because `$pfUTree` will not contain the corresponding item id in the new upload. This can then be used as `$topItem`.